### ISSUE TITLE
Generic struct 'Dictionary' requires that 'RbSymbol' conform to 'Hashable'

### ISSUE THREAD

**Issue by Whirlwind** [johnfairh/RubyGateway#22]
I try to call a ruby method with hash arguments:
```
try Ruby.require(filename: "rubygems")

let gem = try Ruby.get("Gem")
try gem.call("install", args: ["bundler", "> 0", [RbSymbol("bin_dir"): binDir, RbSymbol("install_dir"): gemsDir]])
```
The `RbSymbol` does not conform the `Hashable` protocol.

The ruby api is
```
def self.install(name, version = Gem::Requirement.default, *options)
  bin_dir = options[:bin_dir]
end
```

Please help me.

**Comment by johnfairh** [comment]
Simplest would be:
```swift
let gem = try Ruby.get("Gem")
try gem.call("install",
             args: ["bundler", ">0"],
             kwArgs: ["bin_dir": binDir, "install_dir": gemsDir])
```

**Comment by Whirlwind** [comment]
Thanks, it works.

But If I have a method:
```
def self.install(name, version = Gem::Requirement.default, options1, options2)
  bin_dir = options1[:bin_dir]
  other = options2[:other]
end
```

How to call it? ( there are two hash.)

**Comment by johnfairh** [comment]
To explicitly pass a hash like that you need to build it manually. For example:
```swift
let opts1 = [RbSymbol("bin_dir").rubyObject : binDir]
let opts2 = [RbSymbol("other").rubyObject : otherVal]
try gem.call("install", args: ["bundler", ">0", opts1, opts2])
```

It would be a good enhancement to add `Hashable` to `RbSymbol` to make this a bit simpler like you originally tried to write.

**Comment by Whirlwind** [comment]
Thanks. I known that I could use the `RbSymbol.rubyObject` to get the RbObject which is hashable.
I still hope it could be simpler.

**Comment by johnfairh** [comment]
You're welcome -- I'll look at adding it, won't be very soon though.

**Comment by johnfairh** [comment]
Fixed in 3.1.0.
